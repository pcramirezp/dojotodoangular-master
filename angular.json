{
  "version": 1,
  "projects": {
    "todo": {
      "projectType": "application",
      "schematics": {
        "@nrwl/angular:component": {
          "style": "scss"
        }
      },
      "root": "apps/todo",
      "sourceRoot": "apps/todo/src",
      "prefix": "dojotodoangular",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/apps/todo",
            "index": "apps/todo/src/index.html",
            "main": "apps/todo/src/main.ts",
            "polyfills": "apps/todo/src/polyfills.ts",
            "tsConfig": "apps/todo/tsconfig.app.json",
            "aot": false,
            "assets": ["apps/todo/src/favicon.ico", "apps/todo/src/assets"],
            "styles": ["apps/todo/src/styles.scss"],
            "scripts": []
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "apps/todo/src/environments/environment.ts",
                  "with": "apps/todo/src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "aot": true,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb",
                  "maximumError": "10kb"
                }
              ]
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "todo:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "todo:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "todo:build"
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "apps/todo/tsconfig.app.json",
              "apps/todo/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!apps/todo/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "apps/todo/jest.config.js",
            "tsConfig": "apps/todo/tsconfig.spec.json",
            "setupFile": "apps/todo/src/test-setup.ts"
          }
        }
      }
    },
    "todo-e2e": {
      "root": "apps/todo-e2e",
      "sourceRoot": "apps/todo-e2e/src",
      "projectType": "application",
      "architect": {
        "e2e": {
          "builder": "@nrwl/cypress:cypress",
          "options": {
            "cypressConfig": "apps/todo-e2e/cypress.json",
            "tsConfig": "apps/todo-e2e/tsconfig.e2e.json",
            "devServerTarget": "todo:serve"
          },
          "configurations": {
            "production": {
              "devServerTarget": "todo:serve:production"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/todo-e2e/tsconfig.e2e.json"],
            "exclude": ["**/node_modules/**", "!apps/todo-e2e/**"]
          }
        }
      }
    },
    "data": {
      "root": "libs/data",
      "sourceRoot": "libs/data/src",
      "projectType": "library",
      "schematics": {},
      "architect": {
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/data/tsconfig.lib.json",
              "libs/data/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/data/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/data/jest.config.js",
            "tsConfig": "libs/data/tsconfig.spec.json"
          }
        }
      }
    },
    "ui": {
      "projectType": "library",
      "root": "libs/ui",
      "sourceRoot": "libs/ui/src",
      "prefix": "dojotodoangular",
      "architect": {
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/ui/tsconfig.lib.json",
              "libs/ui/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/ui/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/ui/jest.config.js",
            "tsConfig": "libs/ui/tsconfig.spec.json",
            "setupFile": "libs/ui/src/test-setup.ts"
          }
        },
        "storybook": {
          "builder": "@nrwl/storybook:storybook",
          "options": {
            "uiFramework": "@storybook/angular",
            "port": 4400,
            "config": {
              "configFolder": "libs/ui/.storybook"
            }
          },
          "configurations": {
            "ci": {
              "quiet": true
            }
          }
        },
        "build-storybook": {
          "builder": "@nrwl/storybook:build",
          "options": {
            "uiFramework": "@storybook/angular",
            "outputPath": "dist/storybook/ui",
            "config": {
              "configFolder": "libs/ui/.storybook"
            }
          },
          "configurations": {
            "ci": {
              "quiet": true
            }
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "ui-e2e": {
      "root": "apps/ui-e2e",
      "sourceRoot": "apps/ui-e2e/src",
      "projectType": "application",
      "architect": {
        "e2e": {
          "builder": "@nrwl/cypress:cypress",
          "options": {
            "cypressConfig": "apps/ui-e2e/cypress.json",
            "tsConfig": "apps/ui-e2e/tsconfig.e2e.json",
            "devServerTarget": "ui:storybook"
          },
          "configurations": {
            "production": {
              "devServerTarget": "ui:serve:production"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/ui-e2e/tsconfig.e2e.json"],
            "exclude": ["**/node_modules/**", "!apps/ui-e2e/**"]
          }
        }
      }
    },
    "utils": {
      "root": "libs/utils",
      "sourceRoot": "libs/utils/src",
      "projectType": "library",
      "schematics": {},
      "architect": {
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/utils/tsconfig.lib.json",
              "libs/utils/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/utils/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/utils/jest.config.js",
            "tsConfig": "libs/utils/tsconfig.spec.json"
          }
        }
      }
    }
  },
  "cli": {
    "defaultCollection": "@nrwl/angular"
  },
  "schematics": {
    "@nrwl/angular:application": {
      "unitTestRunner": "jest",
      "e2eTestRunner": "cypress"
    },
    "@nrwl/angular:library": {
      "unitTestRunner": "jest"
    }
  },
  "defaultProject": "todo"
}
