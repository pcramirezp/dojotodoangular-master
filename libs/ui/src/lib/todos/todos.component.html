<span>
    <input type="text" value="Test">
    <div>nuevo elemento <p role="texto">Prueba</p></div>
    
    <button id="boton" onclick="alert('Hola e2e')">Boton</button>
</span>