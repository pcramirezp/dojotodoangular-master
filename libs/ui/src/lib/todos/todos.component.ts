import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'dojotodoangular-todos',
  templateUrl: './todos.component.html',
  styleUrls: ['./todos.component.scss']
})
export class TodosComponent implements OnInit {

  constructor() { }

  ngOnInit() {
  }

}
