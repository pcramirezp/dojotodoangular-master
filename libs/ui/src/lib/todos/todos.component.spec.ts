import { render } from '@testing-library/angular';
import { TodosComponent } from './todos.component';

describe('Suite Todo', () => {
  test('Debe validar que el componente se renderize correctamente', async () => {
    //Arrage
    const RenderResult = await render(TodosComponent, {
      componentProperties: { },
    });

    //Act
    const componentRender = RenderResult.getByRole('texto');

    //Assert
    expect(componentRender);
    expect(componentRender).toMatchSnapshot();
  })
})
