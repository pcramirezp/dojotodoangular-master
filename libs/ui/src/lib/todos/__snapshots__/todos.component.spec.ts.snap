// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suite Todo Debe validar que el componente se renderize correctamente 1`] = `
<p
  role="texto"
>
  Prueba
</p>
`;
